src/Middleware/Diagnostics/src/DummyMeterFactory.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+using Microsoft.Extensions.Diagnostics.Metrics;
+
+namespace Microsoft.AspNetCore.Diagnostics;
+
+internal sealed class DummyMeterFactory : IMeterFactory
```

Comment:
Not needed if you make the public middleware constructor throw.

Reply:
It's not a big deal to add this.

Making the public middleware ctor throw would require breaking change process. I'd prefer to leave it working and go with obsoleting and eventually removing it. Less disruptive.